submodules/power_cloudconnection_create/main.tf

```diff
@@ -103,6 +103,6 @@ resource "ibm_tg_connection" "ibm_tg_connection_2" {
   count        = var.cloud_connection_count > 1 ? 1 : 0
   gateway      = data.ibm_tg_gateway.tg_gateway_ds.id
   network_type = "directlink"
-  name         = "${var.powervs_zone}-conn-2"
+  name         = var.cloud_connection_name_prefix != null && var.cloud_connection_name_prefix != "" ? "${var.cloud_connection_name_prefix}-${var.powervs_zone}-conn-2" : "${var.powervs_zone}-conn-2"
```

Comment:
instead of duplicating this logic so many times, why not add it to a local and reference that?